### PR TITLE
Add line spacing specifications and explanations

### PR DESCRIPTION
**Pull request by ubruhin**
See #29 and #32.

### CODE REVIEW

File: SPECIFICATION.md [hunk 1]
```diff
@@ -224,13 +234,23 @@ parameters:
 2. Global letter spacing (`letter_spacing` from header)
 3. Leading space (leftmost X-coordinate) of second letter
 
+### Line Spacing
+
+Line spacing is the vertical distance between the lines of a multiline text. It
```
Comment (dbrgn) [inline SPECIFICATION.md]: Hm, that's not how line spacing is generally defined. Line spacing (aka leading) is usually baseline to baseline: https://en.wikipedia.org/wiki/Leading I'd stick with that.

File: SPECIFICATION.md [hunk 2]
```diff
@@ -224,13 +234,23 @@ parameters:
...
+can be defined with the `line_spacing` key in the header. The value basically
```
Comment (dbrgn) [inline SPECIFICATION.md]: Can you replace "The value" with "A suitable value"?

File: SPECIFICATION.md [hunk 3]
```diff
@@ -224,13 +234,23 @@ parameters:
...
+depends on the height of ascenders and descenders to avoid vertically
+overlapping characters. So the `line_spacing` value should be set to a value
+slightly greater than the addition of the highest ascender and highest descender
+height. For a typical font this may be around 7 (78% of font height).
```
Comment (dbrgn) [inline SPECIFICATION.md]: If we define the line spacing as baseline to baseline, then we would have to adjust this wording.

From Wikipedia (linked above):

> In both cases, with a 10 pt font the usual leading is 12 pt, creating a 10/12 ratio, which is considered single spacing. Some type designers recommend a leading between 120% and 145% of the font size.

So we could recommend values between 10.8 and 12.6.

**Review by dbrgn — CHANGES_REQUESTED** [review]
I'd really recommend changing the spacing definition from interline spacing to baseline-to-baseline, since that's how it's done in other systems (and in typography in general).

**Comment by ubruhin** [comment]
Thanks for reviewing @dbrgn!

> I'd really recommend changing the spacing definition from interline spacing to baseline-to-baseline

Actually I had it defined like that in the first version, but then I had the idea to define it the same way as letter spacing (for consistency) and because the default value for `line_spacing` would be 0 instead of 9 :sob: 

-> Changed back to baseline distance. OK now?

**Comment by ubruhin** [comment]
> So we could recommend values between 10.8 and 12.6.

That would be far too less. In the Wikipedia article they have another definition for font height (=100%). It seems they count descenders to the font height, which we don't. So with our definition of font height we need much more space. With the "unicode" font I've tested, 178% line distance is fine.

File: SPECIFICATION.md [hunk 4]
```diff
@@ -224,13 +234,24 @@ parameters:
 2. Global letter spacing (`letter_spacing` from header)
 3. Leading space (leftmost X-coordinate) of second letter
 
+### Line Spacing
+
+Line spacing is the vertical distance between the baselines of a multiline text.
+It can be defined with the `line_spacing` key in the header. A suitable value
+basically depends on the height of ascenders and descenders to avoid vertically
```
Comment (dbrgn) [inline SPECIFICATION.md]: Can you remove "basically"? It doesn't add any value.

File: SPECIFICATION.md [hunk 5]
```diff
@@ -236,12 +236,13 @@ parameters:
...
-Line spacing is the vertical distance between the lines of a multiline text. It
-can be defined with the `line_spacing` key in the header. The value basically
-depends on the height of ascenders and descenders to avoid vertically
+Line spacing is the vertical distance between the baselines of a multiline text.
+It can be defined with the `line_spacing` key in the header. A suitable value
+basically depends on the height of ascenders and descenders to avoid vertically
 overlapping characters. So the `line_spacing` value should be set to a value
-slightly greater than the addition of the highest ascender and highest descender
-height. For a typical font this may be around 7 (78% of font height).
+slightly greater than the addition of the cap height (9), highest ascender and
```
Comment (dbrgn) [inline SPECIFICATION.md]: I'd append "height" to the ascender. Yes, it's a repetition, but it's a spec so it's better to stick to more formal language :)

**Review by dbrgn — APPROVED** [review]
Two small wording improvements, otherwise good to merge in my opinion :slightly_smiling_face: Thanks!

**Pull request merged**